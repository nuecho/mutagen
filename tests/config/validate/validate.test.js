const MISSING_DEPENDENCY_CONFIGURATION_PATH = getResourcePath("config/validate/missing-dependency-config.json");
const MULTIPLE_VALIDATION_ERRORS_CONFIGURATION_PATH = getResourcePath("config/validate/multiple-validation-errors-config.json");
const VALID_CONFIGURATION_PATH = getResourcePath("config/validate/valid-config.json");

test(`[mutagen config validate file] should indicate validation failure if there are missing dependencies, and print them`, () => {
  assertMutagenResult(`config validate ${MISSING_DEPENDENCY_CONFIGURATION_PATH}`, "validation failed", 1);
});

test(`[mutagen config validate file] should indicate validation failure if there are multiple validation errors, and print them`, () => {
  assertMutagenResult(`config validate ${MULTIPLE_VALIDATION_ERRORS_CONFIGURATION_PATH}`, "validation failed", 1);
});

test(`[mutagen config validate file] should indicate validation success if there are no validation errors`, () => {
  assertMutagenResult(`config validate ${VALID_CONFIGURATION_PATH}`, "validation succeeded", 0);
});