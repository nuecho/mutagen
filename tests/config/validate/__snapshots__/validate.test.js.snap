// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen config validate file] should indicate validation failure if there are missing dependencies, and print them: validation failed 1`] = `
Object {
  "stderr": "Validation failed.
Missing dependencies:
  - tenant [MissingDependencyTenant1]:
    - script [MissingDependencyTenant1/NonExistingScript1]
  - tenant [MissingDependencyTenant2]:
    - script [MissingDependencyTenant2/NonExistingScript2]
",
  "stdout": "",
}
`;

exports[`[mutagen config validate file] should indicate validation failure if there are multiple validation errors, and print them: validation failed 1`] = `
Object {
  "stderr": "Validation failed.
Missing properties:
  - enumerator [Environment/emptyCfgObjectTest]:
    - displayName
    - type
Missing dependencies:
  - tenant [MissingDependencyTenant1]:
    - script [MissingDependencyTenant1/NonExistingScript1]
  - tenant [MissingDependencyTenant2]:
    - script [MissingDependencyTenant2/NonExistingScript2]
Unchangeable properties:
  - physicalswitch [unchangeablePropertyPhysicalSwitch]:
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen config validate file] should indicate validation success if there are no validation errors: validation succeeded 1`] = `
Object {
  "stderr": "",
  "stdout": "Validation succeeded.
",
}
`;
